src/main/java/noppes/npcs/scripted/ScriptFaction.java

```diff
@@ -99,19 +99,14 @@ public boolean isAggressiveToNpc(ScriptNpc npc){
 		return faction.isAggressiveToNpc(npc.npc);
 	}
 	
-	public int[] getFactionEnemies() {
-		if (faction.attackFactions == null) return null;
-		int[] r = new int[faction.attackFactions.size()];
-		Object[] objects = faction.attackFactions.toArray();
-		for (int i = 0; i < r.length; ++i) {
-			r[i] = (Integer)objects[i];
-		}
-		return r;
+	public Integer[] getFactionEnemies() {
+		if (faction.attackFactions == null) return new Integer[0];
+		return faction.attackFactions.toArray(new Integer[faction.attackFactions.size()]);
 	}
```

Comment:
Have this return an array of IFaction objects instead:

```java
public IFaction[] getFactionEnemies() {
        IFaction[] enemyArr = new IFaction[];
        if (faction.attackFactions == null)
                return enemyArr;
        
        IFactionHandler factionHandler = NpcAPI.Instance().getFactions();
        ArrayList<IFaction> factionList = new ArrayList();
        for (int enemyId : faction.attackFactions) {
                factionList.add(new ScriptFaction(factionHandler.get(enemyId)));
        }
        
        return factionList.toArray();
}

Reply:
I think it makes more sense for you to add this in your pull request because `NpcAPI.Instance().getFactions()` isn't in my branch and my ScriptFaction doesn't implement IFaction.

Reply:
Yep, makes sense! I'll have the function above replace the function that's there now.